.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow builds the docker container and publishes to dockerhub with appropriate tag
+# It has two triggers, 
+#   1. daily i.e. runs everyday at specific time.
+#   2. Anytime a new branch is created under releases
+
+name: Publish Feathr Docker image to DockerHub
+
+on:
+  schedule:
+    # Runs daily at 10 PM UTC, would generate nightly tag
+    - cron: '00 22 * * *'
+  
+  push:
+    # For every push against the releases/** branch, usually would happen at release time, Tag example - releases/v0.7.0
+    branches:
+      - 'releases/**'
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: feathrfeaturestore/feathr-registry
+      
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: FeathrRegistry.Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
```

Comment:
The docker image tags list should be maintained in this workflow internally. Relay on an external list is not reliable and this would make further maintain hard.

Reply:
While I like the idea, it would make the tags a little bit complex since we are supporting two kind of flows through this pipeline and it would require some custom code to decide nightly tag vs release tag which the  docker/metadata-action@v4 action takes care of.